ast/src/analyzed/mod.rs

```diff
@@ -759,6 +759,24 @@ impl Symbol {
         })
     }
 
+    /// Returns an iterator producing either an empty iterator (if it is not an array),
+    /// or all the elements of the array with their names in the form `array[index]`.
+    pub fn array_elements_only(&self) -> impl Iterator<Item = (String, PolyID)> + '_ {
```

Comment:
This function looks a bit too specific for a single use case to me. Instead of calling `array_elements_only`, you could use
```
symbol.is_array().then(|| symbol.array_elements()).flatten()
```